src/main/java/org/kuali/kfs/module/purap/document/web/struts/PurchasingActionBase.java

```diff
@@ -1647,9 +1655,11 @@ private void createReasonNote(ActionForm form) {
     	KualiDocumentFormBase kualiDocumentFormBase = (KualiDocumentFormBase) form;
         try {
             Note noteObj = SpringContext.getBean(DocumentService.class).createNoteFromDocument(kualiDocumentFormBase.getDocument(), ((PurchasingFormBase)kualiDocumentFormBase).getReasonToChange());
+            populateIdentifierOnNoteAndExtension(noteObj);
             kualiDocumentFormBase.getDocument().addNote(noteObj);
-         //   SpringContext.getBean(NoteService.class).save(noteObj);
-            SpringContext.getBean(NoteService.class).saveNoteList(kualiDocumentFormBase.getDocument().getNotes());
+            if (isNoteReadyToBeSaved(kualiDocumentFormBase.getDocument(), noteObj)) {

```

Comment:
I think since this was reverted from the most recent patch that another developer should also review, not because Jay was the original developer for kfspts-9313 but because this wasn't found until in production due to the 'edge case' that caused this to still not work correctly.  I think it is best practice to have at least two developers review this correction.